libraries/chain/include/eosio/chain/controller.hpp

```diff
@@ -355,6 +355,9 @@ namespace eosio { namespace chain {
       void replace_producer_keys( const public_key_type& key );
       void replace_account_keys( name account, name permission, const public_key_type& key );
 
+      void set_producer_node(bool is_producer_node);
+      bool is_producer_node()const;
```

Comment:
Kind of sad about this change. Until now, the core of leap had no need to know if it was a producer or not.

Reply:
well, I guess depending how much of a purist you want to be here, maybe you could make controller take a `std::function<bool(apply_context)> tier_up_with_oc`, either as a ctor parameter or a config or something. That way `chain_plugin` can decide how to set this up based on the various configs. And libtester just always passes in a `return false;` for example, I guess.

Reply:
Yeah, that seems a bit too far.